std/string.d

```diff
@@ -206,23 +206,73 @@ class StringException : Exception
     Params:
         cString = A null-terminated c-style string.
 
-    Returns: A D-style array of `char` referencing the same string.  The
-    returned array will retain the same type qualifiers as the input.
+    Returns: A D-style array of `char`, `wchar` or `dchar` referencing the same
+    string. The returned array will retain the same type qualifiers as the input.
 
     $(RED Important Note:) The returned array is a slice of the original buffer.
     The original data is not changed and not copied.
 +/
 
-inout(char)[] fromStringz(inout(char)* cString) @nogc @system pure nothrow {
-    import core.stdc.string : strlen;
-    return cString ? cString[0 .. strlen(cString)] : null;
+inout(char)[] fromStringz(inout(char)* cString) @nogc @system pure nothrow
```

Comment:
Would deleting this work? We can just rely on the template. Some breakages may occur in odd code that takes address but I think that's broken anyway after adding the template.

Reply:
I argued that before, but the argument was the tests that used string literals break without this. Personally, I think that the tests should just be fixed and the non-templated overload removed. Passing an actual string to `fromStringz` is just broken, and there's no need to pass string literals except for the tests.

Reply:
>I think that's broken anyway after adding the template.

I am happy to remove the old method. The intention was to **not** break old code. IIRC The two testcase breakages were passing `null`, **AND** passing a plain string literal without an `a` suffix i.e "foo", vs "foo"a

Reply:
We don't want to break code, but given that the entire point of `fromString` is to take a null-terminated string and convert it to a D string, I don't see how it's a valid use case to pass it either a string literal or a `null` literal. They might have made sense as a simple way to write a test, but it's not something that real code would do.

Reply:
>I would be very surprised if anyone actually uses fromStringz with a string literal. It's completely pointless.

@jmdavis Yes this was the jist of your argument before. I completely agree, but I was leaving it up to someone to provide a definitive decision.